src/com/ensa/gi4/datatabase/impl/GenericDAO.java

```diff
@@ -5,26 +5,80 @@
 import org.springframework.jdbc.core.JdbcTemplate;
 import org.springframework.jdbc.core.RowMapper;
 
+import com.ensa.gi4.modele.Materiel;
+
 import javax.sql.DataSource;
 import java.util.List;
 
-public abstract class GenericDAO<T> implements InitializingBean {
-    @Autowired
-    private DataSource dataSource;
-    private JdbcTemplate jdbcTemplate;
+public abstract class GenericDAO implements InitializingBean {
+	@Autowired
+	private DataSource dataSource;
+	private JdbcTemplate jdbcTemplate;
+
+	@Override
+	public void afterPropertiesSet() { // from InitializingBean
+		jdbcTemplate = new JdbcTemplate(dataSource);
+	}
+
+	protected void add(String table, Materiel materiel) {
+		String query = "INSERT INTO " + table + " (NAME,CODE,STOCK,DISPONIBILITE) VALUES (?,?,?,?)";
```

Comment:
concatenate strings while building a sql query is dangerous, this opens the door easily to an SqLInjection

Reply:
ok